crates/oxc_prettier/src/lib.rs

```diff
@@ -21,6 +27,25 @@ use crate::{doc::Doc, format::Format, printer::Printer};
 
 pub use crate::options::{ArrowParens, EndOfLine, PrettierOptions, QuoteProps, TrailingComma};
 
+pub struct UniqueGroupIdBuilder {
+    next_id: AtomicU32,
+}
+
+impl Default for UniqueGroupIdBuilder {
+    fn default() -> Self {
+        Self { next_id: AtomicU32::new(1) }
+    }
+}
+
+impl UniqueGroupIdBuilder {
+    pub fn group_id(&self) -> GroupId {
+        let id = self.next_id.fetch_add(1, Ordering::Relaxed);
+        dbg!(id);
+        let id = NonZeroU32::new(id).unwrap_or_else(|| panic!("Group ID counter overflowed"));
+
+        GroupId::new(id)
+    }
+}
```

Comment:
think we can simplify this - don't think it needs to be send/sync

Reply:
What's happening here? Everything should be single threaded 🤔 You can `Rc<RefCell>>` it if you need inner mutability.